include/aikido/control/ros/RosJointStateClient.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef AIKIDO_CONTROL_ROS_ROSJOINTSTATECLIENT_HPP_
+#define AIKIDO_CONTROL_ROS_ROSJOINTSTATECLIENT_HPP_
+#include <string>
+#include <boost/circular_buffer.hpp>
+#include <dart/dynamics/dynamics.hpp>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <sensor_msgs/JointState.h>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+class RosJointStateClient
+{
+public:
+  RosJointStateClient(
+    dart::dynamics::SkeletonPtr _skeleton,
+    ::ros::NodeHandle _nodeHandle,
+    const std::string& _topicName,
+    size_t capacity);
+
+  void spin();
+
+  Eigen::VectorXd getLatestPosition(
+    const dart::dynamics::MetaSkeleton& _metaSkeleton) const;
```

Comment:
We should add a second functions as well:
```c++
Eigen::VectorXd getPositionAtTime(const MetaSkeleton&, const ros::Time&, bool)
```
that interpolates position at the specified time - optionally blocking for new data. This would eliminate a *huge* class of race conditions that we see in OpenRAVE.

However, this is purely an addition. So we can put this off for a follow-up pull request.